library/core/resource-loader/src/main/java/org/quiltmc/qsl/resource/loader/api/GroupResourcePack.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.resource.loader.api;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
+
+import net.minecraft.resource.ResourceNotFoundException;
+import net.minecraft.resource.ResourcePack;
+import net.minecraft.resource.ResourceType;
+import net.minecraft.util.Identifier;
+
+/**
+ * Represents a group resource pack, which holds multiple resource packs as one.
+ */
+public abstract class GroupResourcePack implements ModResourcePack {
```

Comment:
Hmm what was the purpose of this again? Might be worth describing the use case in docs

Reply:
In FAPI it's impl-only, but the goal of this is to be able to group multiple resource packs as one unique entry in the resource pack list.

Which will be used for mod resource packs and vanilla resource pack, as it'll be one unique entry (Default) in the resource pack screen.